mpqp/core/circuit.py

```
@@ -345,23 +355,33 @@ def _update_targets_components(self, component: Instruction | NoiseModel):
                 c_targets.append(i)
                 i += 1
             component.c_targets = c_targets
-            self.nb_cbits = max(
-                max(c_targets, default=0) + 1, max(unique_cbits, default=0) + 1
-            )
 
+            self._update_cbits(
+                max(max(c_targets, default=0) + 1, max(unique_cbits, default=0) + 1)
+            )
         return component
 
     @property
     def nb_qubits(self) -> int:
         """Number of qubits of the circuit."""
         return self._nb_qubits if self._user_nb_qubits is None else self._user_nb_qubits
 
+    @property
+    def nb_cbits(self) -> int:
+        """Number of cbits of the circuit."""
+        return self._nb_cbits if self._user_nb_cbits is None else self._user_nb_cbits
+
     @nb_qubits.setter
     def nb_qubits(self, nb_qubits: int):
         if self._user_nb_qubits is None or self._user_nb_qubits != nb_qubits:
             self._user_nb_qubits = nb_qubits
             self._set_nb_qubits_dynamic(nb_qubits)
 
+    @nb_cbits.setter
+    def nb_cbits(self, nb_cbits: int):
+        if self._user_nb_cbits is None or self._user_nb_cbits != nb_cbits:
```

If `nb_cbits < _user_nb_cbits` we should return an error right ?

---

ok for me (error message cleaned in commit 84bef)